ViewManagement/Initialize.lua

```diff
@@ -224,18 +224,25 @@ local function SetupBankView()
     ResetPositions()
   end)
 
-  addonTable.CallbackRegistry:RegisterCallback("BankToggle", function(_, characterName)
-    characterName = characterName or Syndicator.API.GetCurrentCharacter()
-    bankView:SetShown(characterName ~= bankView.Character.lastCharacter or not bankView:IsShown())
-    bankView:UpdateViewToCharacter(characterName)
+  addonTable.CallbackRegistry:RegisterCallback("BankToggle", function(_, entity, subView)
+    if type(entity) == "string" or entity == nil then -- Character bank
+      local characterName = entity or Syndicator.API.GetCurrentCharacter()
+      bankView:SetShown(characterName ~= bankView.Character.lastCharacter or not bankView:IsShown())
+      bankView:UpdateViewToCharacter(characterName)
+    elseif type(entity) == "number" then -- Warband bank
+      subView = subView or addonTable.Config.Get(addonTable.Config.Options.WARBAND_CURRENT_TAB)
+      bankView:SetShown(not bankView:IsShown())
+      bankView:UpdateViewToWarband(entity, subView)
+    end
   end)
 
   addonTable.CallbackRegistry:RegisterCallback("BankShow", function(_, entity, subView)
-    if type(entity) == "string" or entity == nil then -- Character bank
+    local selectedTab = GetSelectedBankTab(entity)
+    if selectedTab == characterTab then -- Character bank
       local characterName = entity or Syndicator.API.GetCurrentCharacter()
       bankView:Show()
       bankView:UpdateViewToCharacter(characterName)
-    elseif type(entity) == "number" then -- Warband bank
+    elseif selectedTab == warbandTab then -- Warband bank
```

Comment:
This looks broken now that the GetSelectedBankTab function is gone

Reply:
Do you prefer the logic to remain in `GetSelectedBankTab()` for both `BankToggle` and `BankShow`, or just roll it back to what it was before using if statements in each?

Reply:
Mostly. Can you use a structure like https://github.com/Baganator/Baganator/blob/main/Core/Constants.lua#L80-L85 to hold the constants for the tabs (changing them to labelled numbers), rather than using the string values.

Reply:
Place the new structure on the addonTable in Constants.lua below everything else

Reply:
Something like this?
```
addonTable.Constants.TabTypes = {
  Character = 0,
  Warband = 1,
}
```

Reply:
Call it BankTabType but otherwise its good

Reply:
done 😄